lib/teach-api.js

```diff
@@ -38,8 +38,13 @@ util.inherits(TeachAPI, EventEmitter);
 
 _.extend(TeachAPI.prototype, {
   logout: function() {
+    var currentUsername = this.getUsername();
+
     delete this.storage[STORAGE_KEY];
-    this.emit('username:change', null);
+
+    if (currentUsername !== null) {
```

Comment:
I'm kinda confused here - how could a user log out if he/she hasn't logged in yet?